### PR TITLE
feat: implement plug agent

### PR DESCRIPTION
**Pull request by aleacevedo**
## Changelog

- Change 1
- Change 2
- Change 3

### Type of changes included:

- [ ] Components
- [x] Business Logic
- [ ] Bug fixes
- [ ] Styling
- [ ] Code Refactor

### Clubhouse Tickets Related:

- Ticket 1
- Ticket 2
- Ticket 3

### Screenshots/Gifs:

*Place you changes screenshots here*

### Notes:

*Place special notes here*

### Tested on:

- [ ] Chrome
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

File: source/Background/utils.js [hunk 1]
```diff
@@ -106,9 +107,108 @@ export const fetchCanistersInfo = async (whitelist) => {
...
+    );
+
+    if (!allArgsEqual) {
+      return false;
+    }
+  }
 
-  return assets.filter(asset => asset.symbol === tokenIdentifier)[0]
-}
+  return true;
```
Comment (rocky-fleek) [inline source/Background/utils.js:145]: you can do `return allArgsEqual`

File: source/Background/utils.js [hunk 2]
```diff
@@ -106,9 +107,108 @@ export const fetchCanistersInfo = async (whitelist) => {
...
+};
+
+export const handleCallRequest = async ({
```
Comment (rocky-fleek) [inline source/Background/utils.js:148]: This doesnt seem to belong here, I'd put it in the transactions module right?

File: source/Modules/Controller/transaction.js [hunk 3]
```diff
@@ -401,72 +440,176 @@ export class TransactionModule extends ControllerModuleBase {
...
           const isDangerousUpdateCall = !preApprove && requestType === 'call';
           if (isDangerousUpdateCall) {
-            getApps(this.keyring?.currentWalletId.toString(), async (apps = {}) => {
+            getApps(
+              this.keyring?.currentWalletId.toString(),
+              async (apps = {}) => {
+                const app = apps?.[metadata.url] || {};
+                if (app.status !== CONNECTION_STATUS.accepted) {
+                  callback(ERRORS.CONNECTION_ERROR, null);
+                  return;
```
Comment (rocky-fleek) [inline source/Modules/Controller/transaction.js:449]: use `getApp`

File: source/Background/utils.js [hunk 4]
```diff
@@ -106,9 +107,108 @@ export const fetchCanistersInfo = async (whitelist) => {
...
 // Return ICP by default
 export const getToken = (tokenIdentifier, assets) => {
-  if (!tokenIdentifier) return assets.filter(asset => asset.canisterId === ICP_CANISTER_ID)[0]
+  if (!tokenIdentifier) return assets.filter((asset) => asset.canisterId === ICP_CANISTER_ID)[0];
+
+  if (validateCanisterId(tokenIdentifier)) {
+    return assets.filter((asset) => asset.canisterId === tokenIdentifier)[0];
+  }
+
+  return assets.filter((asset) => asset.symbol === tokenIdentifier)[0];
```
[the remark this answers is not in the document]
Comment (aleacevedo) [reply source/Background/utils.js:116]: Nop, only in the case where tokenIdentifier is undefined it should return ICP asset in any other case it should return the asset of tokenIdentifier

File: source/Background/utils.js (hunk 2 above)
Comment (aleacevedo) [reply source/Background/utils.js:148]: Not sure, I would put only RPC handlers into the modules

File: source/Background/utils.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (aleacevedo) [reply source/Background/utils.js]: yep actually u are right

**Review by tomiir — APPROVED** [review]

**Pull request merged**